### DISCUSSION TITLE
Flutter and Web3Wallet

### DISCUSSION THREAD

**Discussion by marcomanzi** [WalletConnect/walletconnect-monorepo#2337] [Developer Support] [answered]
I'm trying to integrate Web3Wallet on web application written in flutter, I was able to generate the QR code but I can't connect with Trust wallet. 
I see that the guide is very cofused on this simple task, can anyone give me a simple example of this?

**Comment by TheAmjadKhan** [comment, marked as answer]
which package are you using? 

I am using https://pub.dev/packages/walletconnect_flutter_v2 and it's working perfectly

if you are using same package 
when you show QR code you have to listen and wait for response 

```final sessionData = await res.session.future;```

**Comment by finessevanes** [comment]
@marcomanzi can you provide your code implementation?

**Comment by bobwith2bees** [comment]
@marcomanzi  
An additional Flutter app using the https://pub.dev/packages/walletconnect_flutter_v2  package with full code (more focused on Android/IOS than web app at the moment):

https://gitlab.com/graflr/flutter_web3_demo

**Reply by imperialrogers** [reply]
there is error in the explorer_registry_listing.dart file in the gitlab link under models folder in lib showing that  the 5th line part 'explorer_resgistry_listing.g.dart'
Target url hasnt been generated

**Reply by bobwith2bees** [reply]
@imperialrogers  - for the class I am having student generate the abi code and models.  See the readme on how to create the .g.dart files.

https://gitlab.com/graflr/flutter_web3_demo#generate-abi-wrappers-that-go-in-libabi

And a reminder - my example code (the flutter_web3_demo)  is not part of this repo, or the official project.      I should have opened the issues feature on that project if you have more trouble or questions on that.   Not sure this is the best forum for supporting my code ;-)

**Reply by finessevanes** [reply]
@bobwith2bees you can always add your project to [WalletConnect Awesome](https://github.com/WalletConnect/awesome-walletconnect)!

**Comment by finessevanes** [comment]
@Ma
